### ISSUE TITLE
Idea to simplify Windows build process

### ISSUE THREAD

**Issue by pal1000** [PCSX2/pcsx2#418]
Considering that building PCSX2 from source demands DirectX SDK to be installed and it's path need to be set at project level as Visual Studio deprecated global paths settings this raise development complexity in the following ways;
-DirectX SDK path needs to be set on multiple projects: Gsdx, LilyPad, spu2- x and xpad;
-this modifies git tracked files - these changes needs to be stashed from actual development branch before pushing it to personal github forked repository;

Solution: Submodule.
@hrydgard was very smart when he created this submodule for PPSSPP. The nice part is that PCSX2 may use it too.
Repository location: https://github.com/hrydgard/minidx9
This would make the paths to be static and as result PCSX2 build process would be much easier for those who try to do it for the first time.

**Comment by ramapcsx2** [comment]
I'm not quite sure the Microsoft license allows doing that. Did someone review it for PPSSPP?
We would need this for DX9 and DX10/11, too.
Otherwise I like this. The usual workaround really is annoying, especially for first timers.

**Comment by hrydgard** [comment]
Honestly I'm not completely sure that's it's 100% kosher, but it's just a few headers and lib files that are freely available anyway, just packaged differently, and nobody has complained so far. Worth that very small worry for the convenience. It doesn't affect the licensing of the final product anyway.

**Comment by pal1000** [comment]
I built PCSX2 using that package. Everything worked just fine as soon as I collected info on all the projects demanding DirectX SDK. The only thing that could not be compiled was the xpad plugin. Don't have an Xbox 360 controller so I don't depend on that anyway.
I built it using Release AVX - Win32 profile.
Regarding why xpad build failed here is the output:
![temp](https://cloud.githubusercontent.com/assets/1138235/5614362/15109da0-94f9-11e4-94cf-93519f6320a3.png)

**Comment by micove** [comment]
This could be added to a global prop or made into a new prop like this:
```
<?xml version="1.0" encoding="utf-8"?>
<Project ToolsVersion="4.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
  <ImportGroup Label="PropertySheets" />
  <PropertyGroup Label="UserMacros" />
  <PropertyGroup />
  <ItemDefinitionGroup Condition="'$(ProjectName)'=='SPU2-X' or '$(ProjectName)'=='GSdx' or '$(ProjectName)'=='LilyPad'">
    <ClCompile>
      <AdditionalIncludeDirectories>$(DXSDK_DIR)include;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
    </ClCompile>
    <Link>
      <AdditionalLibraryDirectories Condition="'$(Platform)'=='Win32'">$(DXSDK_DIR)Lib\x86;%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
      <AdditionalLibraryDirectories Condition="'$(Platform)'=='x64'">$(DXSDK_DIR)Lib\x64;%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
    </Link>
  </ItemDefinitionGroup>
  <ItemGroup />
</Project>
```

Then if the user selected to not create $(DXSDK_DIR) then it only needs to be changed in only one place. This avoid possible EULA issues.

The EULA says that you can only distribute these:
```
i.	Right to Use and Distribute.  The code and text files listed below are Distributable Code.
	DIRECTX REDIST.TXT Files.  You may copy and distribute the object code form of code listed in DIRECTX REDIST.TXT files.
	Sample Code.  You may modify, copy, and distribute the source and object code form of code marked as sample, as well as those marked as follows:
\Utilities\bin\x86\dxerr
\Utilities\bin\x64\dxerr
\Utilities\bin\x86\dxtex
\Utilities\bin\x64\dxtex
\Utilities\bin\x86\DxViewer
\Utilities\bin\x64\DxViewer
\Utilities\bin\x86\GDFTrace
\Utilities\bin\x64\GDFTrace
\Utilities\bin\x86\MeshConvert
\Utilities\bin\x64\MeshConvert
\Utilities\Source\Sas
\Utilities\Source\Effects11
```
and the DIRECTX REDIST.TXT says
```
DirectX Software Development Kit Redistributable Code
-------------------------------------------------------------
Redistributable Code is identified as all of the files in the following 
"Redist" directory on the DirectX SDK CD/Download:

   redist/


The following files in the above directory must be included in any distribution of 
the DirectX Runtime.

DSetup32.dll
DSetup.dll
DXSetup.exe
DXupdate.cab
dxdllreg_x86.cab
```

It seems to say that the headers and dev libraries are not included in the things that you may distribute.

**Comment by pal1000** [comment]
Well, I guess I am willing to close this when all projects needing DirectX SDK will detect DXSDK_DIR variable in all build modes currently supported: Release, Devel and Debug.
I did a through digging on these projects testing which can and which can't detect DXSDK_DIR variable and here are the results: 
-spu2-x is OK. It finds it in all modes: Release, Devel and Debug;
-LilyPad only finds it in Release mode (all CPU specific sub-mods);
-GSdx does not find it all!
-xpad is broken severely. Even if you set paths properly it would still be unable to find XInputEnable definition.  


I think it's enough to have them detect DXSDK_DIR.

**Comment by Pistachioman** [comment]
What do you mean by "detect" the variable? I included it in all necessary project in my branch and it works without problems (shameless plug for #233)

**Comment by pal1000** [comment]
I mean. for example GSdx cannot find DirectX SDK path via DXSDK_DIR and as such it fails compilation on a clean copy of master branch. Have to manually edit include and libraries directories in Project properties - VC++ Directories by appending <code>$(DXSDK_DIR)\Include;</code> and <code>$(DXSDK_DIR)\Lib\x86;</code> respectively to include directories and libraries directories to compile successfully.
I made some tests and determined which projects requiring DirectX SDK are unable to find it even if DXSDK_DIR is set properly and in which compilation scenarios. From your reply I understand you already took note of them.

**Comment by micove** [comment]
Wild guess but did you set the system  environment variable as
DXSDK_DIR -> "C:\Program Files\Microsoft DirectX SDK (June 2010)\"
I mean with the trailing "\"

The official installer creates that system environment variable with a trailing backslash. If don't use the official installer and use something like from the git link you posted then you need to make sure you add the trailing backslash since:

https://github.com/PCSX2/pcsx2/blob/master/plugins/GSdx/vsprops/common.props
https://github.com/PCSX2/pcsx2/blob/master/plugins/GSdx/vsprops/x64.props
https://github.com/PCSX2/pcsx2/blob/master/plugins/GSdx/vsprops/x86.props

Already include the DXSDK line as $(DXSDK_DIR)Lib and $(DXSDK_DIR)include which is the normal convention. Since you posted that it works as $(DXSDK_DIR)\Include and $(DXSDK_DIR)\Lib  then I can guess that you just need to update the variable and it should work.

**Comment by avih** [comment]
FWIW, I recently setup a new windows 8.1 system from svratch, with VS2013 and the DX_SDK, and it built right out of the box without any project tinkering. See http://forums.pcsx2.net/Thread-GIT-build-crashes-when-trying-to-load-game?pid=427541#pid427541

**Comment by pal1000** [comment]
@micove already helped me with this, that's why I closed it. I wanted to cheat the installer to give me only the headers and libraries without silently installing anything. This is realistic and can be done if you know what you are doing. But I screwed up, not paying attention at the format of DXSDK_DIR variable value while testing the installer on a virtual machine.
I updated the wiki article about compilation on Windows based on my findings.
http://wiki.pcsx2.net/index.php/Compiling_on_Windows#Installing_SDKs.
